joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/rewrite/RewriteAutoConfiguration.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2016-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.joinfaces.autoconfigure.rewrite;
+
+
+import javax.inject.Inject;
+
+import org.ocpsoft.rewrite.servlet.RewriteFilter;
+import org.ocpsoft.rewrite.servlet.impl.RewriteServletContextListener;
+import org.ocpsoft.rewrite.servlet.impl.RewriteServletRequestListener;
+
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.boot.web.servlet.ServletListenerRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.DependsOn;
+
+/**
+ * Spring Boot Auto Configuration of Rewrite.
+ *
+ * @author Marcelo Fernandes
+ */
+@Configuration
+@EnableConfigurationProperties({RewriteFilterProperties.class, RewriteProperties.class})
+@ConditionalOnClass(RewriteFilter.class)
+@AutoConfigureAfter(WebMvcAutoConfiguration.class)
+@ConditionalOnWebApplication(type = ConditionalOnWebApplication.Type.SERVLET)
+public class RewriteAutoConfiguration {
```

Comment:
The listeners and filters here will collide with `rewrite-servlet-3.4.2.Final.jar!/META-INF/web-fragment.xml` when used with war-deployment.

Reply:
You are right. Do you know how to detect war-deployment to avoid it?

Reply:
I fixed this in e9f563b

When deploying the application as war file to an existing container, the existing container will read the `web-fragment.xml`. When using and embedded servlet container, the `WebServerFactoryCustomizer` will do the same as the `web-fragment.xml`-file